dist/twine.js

```diff
@@ -1,537 +1,509 @@
-(function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module '"+o+"'");throw f.code="MODULE_NOT_FOUND",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){
-var attribute, bind, currentBindingCallbacks, elements, eventName, fireCustomChangeEvent, getContext, getValue, isKeypath, j, k, keypathForKey, keypathRegex, len, len1, nodeCount, preventDefaultForEvent, ref, ref1, refreshElement, refreshQueued, rootContext, rootNode, setValue, setupAttributeBinding, setupEventBinding, stringifyNodeAttributes, valueAttributeForNode, wrapFunctionString,
-  slice = [].slice;
-
-window.Twine = {};
-
-Twine.shouldDiscardEvent = {};
-
-elements = {};
-
-nodeCount = 0;
-
-rootContext = null;
-
-keypathRegex = /^[a-z]\w*(\.[a-z]\w*|\[\d+\])*$/i;
-
-refreshQueued = false;
-
-rootNode = null;
-
-currentBindingCallbacks = null;
-
-Twine.getAttribute = function(node, attr) {
-  return node.getAttribute("data-" + attr) || node.getAttribute(attr);
-};
-
-Twine.reset = function(newContext, node) {
-  var bindings, j, key, len, obj, ref;
-  if (node == null) {
-    node = document.documentElement;
-  }
-  for (key in elements) {
-    if (bindings = (ref = elements[key]) != null ? ref.bindings : void 0) {
-      for (j = 0, len = bindings.length; j < len; j++) {
-        obj = bindings[j];
-        if (obj.teardown) {
-          obj.teardown();
+(function() {
+  var slice = [].slice;
+
+  (function(root, factory) {
+    if (typeof root.define === 'function' && root.define.amd) {
+      return root.define;
```

Comment:
Is this correct for AMD? The [pattern defined on the github repo](https://github.com/umdjs/umd/blob/master/templates/returnExports.js#L44) seems to call `define([], factory)`